src/Sylius/Behat/Context/Setup/Api/CartContext.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\Behat\Context\Setup\Api;
+
+use Behat\Behat\Context\Context;
+use Sylius\Behat\Client\ApiClientInterface;
+use Sylius\Behat\Client\Request;
+use Sylius\Behat\Client\ResponseCheckerInterface;
+use Sylius\Behat\Service\SharedStorageInterface;
+use Sylius\Component\Core\Model\ProductInterface;
+use Sylius\Component\Core\Model\ProductVariantInterface;
+use Sylius\Component\Product\Model\ProductOptionInterface;
+use Sylius\Component\Product\Model\ProductOptionValueInterface;
+use Sylius\Component\Product\Resolver\ProductVariantResolverInterface;
+use Symfony\Component\HttpFoundation\Request as HttpRequest;
+
+final class CartContext implements Context
+{
+    /** @var ApiClientInterface */
+    private $cartsClient;
+
+    /** @var ResponseCheckerInterface */
+    private $responseChecker;
+
+    /** @var SharedStorageInterface */
+    private $sharedStorage;
+
+    /** @var ProductVariantResolverInterface */
+    private $productVariantResolver;
+
+    public function __construct(
+        ApiClientInterface $cartsClient,
+        ResponseCheckerInterface $responseChecker,
+        SharedStorageInterface $sharedStorage,
+        ProductVariantResolverInterface $productVariantResolver
+    ) {
+        $this->cartsClient = $cartsClient;
+        $this->responseChecker = $responseChecker;
+        $this->sharedStorage = $sharedStorage;
+        $this->productVariantResolver = $productVariantResolver;
+    }
+
+    /**
+     * @Given /^I (?:have|had) (product "[^"]+") in the (cart)$/
+     * @Given /^I (?:have|had) (product "[^"]+") added to the (cart)$/
+     * @Given /^I added (product "[^"]+") to the (cart)$/
+     * @Given /^the (?:customer|visitor) has (product "[^"]+") in the (cart)$/
+     * @When /^the (?:customer|visitor) try to add (product "[^"]+") in the customer (cart)$/
+     */
+    public function iAddedProductToTheCart(ProductInterface $product, ?string $tokenValue): void
```

Comment:
In my opinion, in setup, we should dispatch commands instead of making requests as it was already done. Why did you change that in this PR?

Reply:
Dispatching commands was not setting correct hostname to make requests by API. They were called (API calls) to lets say 'default.com' not 'usa-store.com' hence it generate the problem of creating order in wrong channel.

Reply:
We are hitting the problems of stateful commands. ChannelContext is taking the default channel (or not at all), so our pickup-cart logic begun to be broken. Removing coupling of PickupCartHandler with ChannelContext and moving channel itself to the command would allow us to keep the current channel in shared storage and use it to pick a cart. On the command implementation, I think it should be part of data transformers (similar to order token value, but here we can use context)